Dockerfile

```
@@ -0,0 +1,63 @@
+FROM mcr.microsoft.com/vscode/devcontainers/base:0.202.7-bullseye
+ARG USER=vscode
+ARG UID=1000
+ARG GID=${UID}
+ARG NIX_INSTALLER=https://releases.nixos.org/nix/nix-2.10.3/install
+ARG CHANNEL_URL=https://github.com/NixOS/nixpkgs/archive/aaa1c973c8c189195e1b1a702d3b74dbcde91538.tar.gz
+ARG CACHIX_NAME=composable-community
+
+SHELL [ "/bin/bash", "-o", "pipefail", "-o", "errexit", "-c" ]
+
+RUN export DEBIAN_FRONTEND=noninteractive && \
+    apt-get update && \
+    apt-get install --yes --no-install-recommends \
+    ca-certificates \
+    curl \
+    sudo \
+    xz-utils \
+    acl
+
+RUN addgroup --system nixbld
+RUN for i in $(seq 1 30); do useradd -ms /bin/bash nixbld$i &&  adduser nixbld$i nixbld; done
+RUN adduser ${USER} nixbld
+
+RUN usermod --append --groups sudo ${USER} --shell /bin/bash
+RUN adduser ${USER} root
+RUN sed --in-place 's/%sudo.*ALL/%sudo   ALL=(ALL:ALL) NOPASSWD:ALL/' /etc/sudoers
+
+RUN mkdir --parents /etc/nix/
+RUN echo "sandbox = relaxed" >> /etc/nix/nix.conf
+RUN echo "experimental-features = nix-command flakes" >> /etc/nix/nix.conf
+RUN echo "narinfo-cache-negative-ttl = 30" >> /etc/nix/nix.conf 
+RUN curl --location ${NIX_INSTALLER} > /install.sh
+RUN chmod +x /install.sh
+RUN passwd --delete root
+
+USER ${USER}
+ENV USER=${USER}
+
+RUN /install.sh
```

you download install.sh above, but use it here
```suggestion
RUN echo "narinfo-cache-negative-ttl = 30" >> /etc/nix/nix.conf 
RUN passwd --delete root

USER ${USER}
ENV USER=${USER}

RUN curl --location ${NIX_INSTALLER} > /install.sh && \
         chmod +x /install.sh  && \
         /install.sh
```

---

yes. its better.